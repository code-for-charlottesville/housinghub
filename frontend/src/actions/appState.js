export function setLoading(loading) {
  return {
    type: "SET_LOADING",
    loading,
  };
}

export function setSidebarOpen(sidebarOpen) {
  return {
    type: "SET_SIDEBAR_OPEN",
    sidebarOpen,
  };
}
