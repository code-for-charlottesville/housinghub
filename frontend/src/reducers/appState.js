const initialState = {
  loading: false,
};

const appState = (state = initialState, action) => {
  switch (action.type) {
    case "SET_LOADING":
      return Object.assign({}, state, {
        ...state,
        loading: action.loading,
      });
    default:
      return state;
  }
};

export default appState;
