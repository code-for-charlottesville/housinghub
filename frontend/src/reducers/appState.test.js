import appState from "./appState";

describe("reducers/appState", () => {
  let testTable = [
    {
      name: "sets loading",
      action: {
        type: "SET_LOADING",
        loading: true,
      },
    },
    {
      name: "set sidebar open",
      action: {
        type: "SET_SIDEBAR_OPEN",
        sidebarOpen: true,
      },
    },
  ];
  testTable.forEach((t) => {
    it(t.name, () => {
      expect(appState(undefined, t.action)).toMatchSnapshot();
    });
  });
});
