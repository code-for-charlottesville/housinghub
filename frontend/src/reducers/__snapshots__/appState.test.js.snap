// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/appState set sidebar open 1`] = `
Object {
  "loading": false,
  "sidebarOpen": true,
}
`;

exports[`reducers/appState sets loading 1`] = `
Object {
  "loading": true,
  "sidebarOpen": false,
}
`;
